frb_codegen/src/generator/rust/ty_delegate.rs

```diff
@@ -161,3 +197,25 @@ impl TypeRustGeneratorTrait for TypeDelegateGenerator<'_> {
         delegate_enum!(self, static_checks(), None)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn io_wire2api() {
+        // input in microseconds
+        let input: i64 = 3_496_567_123;
+        let s = (input / 1_000_000) as i64;
+        let ns = (input.rem_euclid(1_000_000) * 1_000) as u32;
+        assert_eq!(s, 3_496);
+        assert_eq!(ns, 567_123_000);
+    }
+    #[test]
+    fn wasm_wire2api() {
+        // input in milliseconds
+        let input: i64 = 3_496_567;
+        let s = (input / 1_000) as i64;
+        let ns = (input.rem_euclid(1_000) * 1_000_000) as u32;
+        assert_eq!(s, 3_496);
+        assert_eq!(ns, 567_000_000);
+    }
+}
```

Comment:
Tests are great, but they do not (cannot indeed) test the to-be-generated code.

Thus, what about moving *everything* into frb_dart and frb_rust, instead of being a lot of literal strings? Then we can happily test them :)

Reply:
@fzyzcjy Do you mean like some `extension` on Dart type and `impl` on Rust type ?

Reply:
No, just plain functions in frb_dart and frb_rust are enough.

After write down those functions, we can easily unit test them as well

Reply:
done for rust in c4d1ef70954c9e86708fac67a64c6ac19fff6929

Reply:
on Dart side there's not much, can we just skip it ?

Reply:
ah no my bad, there's some sorry.

Reply:
haha actually I got confused well there's mostly `DateTime.fromMillisecondsSinceEpoch` or `fromMicrosecondsSinceEpoch`, so yes should I refactor them ?

Reply:
ok so done on dart side with e2911f60236663dcf36348a2a74b96470d89dbca